toolchain/sem_ir/value_stores.h

```diff
@@ -53,6 +53,49 @@ class InstStore {
   ValueStore<InstId> values_;
 };
 
+// Provides a ValueStore wrapper for tracking the constant values of
+// instructions.
+class ConstantValueStore : public Yaml::Printable<ConstantValueStore> {
+ public:
+  // Returns the constant value of the requested instruction, or InstId::Invalid
+  // if it is not constant.
+  auto Get(InstId inst_id) const -> InstId {
+    CARBON_CHECK(inst_id.index >= 0);
+    return static_cast<size_t>(inst_id.index) >= values_.size()
+               ? InstId::Invalid
+               : values_[inst_id.index];
+  }
+
+  // Sets the constant value of the given instruction.
+  auto Set(InstId inst_id, InstId const_id) -> void {
+    CARBON_CHECK(inst_id.index >= 0);
+    if (values_.size() <= static_cast<size_t>(inst_id.index)) {
```

Comment:
Sure, done.

(FWIW, for me the difference between these two is that the line 64 comparison is saying "if the index is out of range, then use this other value", and this case is saying "if the size is insufficient, then change it" -- so that the "object" of the comparison is on the left and the "subject" is on the right -- but it seems equally clear to me here to say "if the index doesn't fit in the container then resize the container so it fits".)